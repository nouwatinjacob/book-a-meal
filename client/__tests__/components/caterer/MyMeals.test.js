import React from 'react';
import {
  shallow
} from 'enzyme';
import {
  MyMeals,
  mapStateToProps
} from '../../../src/components/caterer/MyMeals.jsx';
import {
  allMealResponse,
} from '../../__mockData__/mockMeal';

const props = {
  deleteMealAction: () => Promise.resolve(),
  getMeals: () => Promise.resolve(),
  mealState: {
    allMealResponse,
    paginate: allMealResponse.paginate
  },
};

describe('MyMeals component', () => {
  localStorage.setItem('token', catererToken);
  const setup = () => shallow(<MyMeals {...props} />);

  it('should render MyMeals component correctly', (done) => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    done();
  });

  it(`should call 'handlePageClick'
  if pagination button is clicked`, (done) => {
    const wrapper = setup();

    const data = {
      selected: 1
    };

    const handlePageClickSpy = jest.spyOn(
      wrapper.instance(),
      'handlePageClick'
    );
    wrapper.instance().handlePageClick(data);
    expect(handlePageClickSpy).toHaveBeenCalled();

    done();
  });

  it(`should call 'handleDelete'
  if delete button is clicked`, (done) => {
    const wrapper = setup();

    const mealId = 1;
    window.location.reload = jest.fn();
    const handleDeleteSpy = jest.spyOn(
      wrapper.instance(),
      'handleDelete'
    );
    wrapper.instance().handleDelete(mealId);
    expect(handleDeleteSpy).toHaveBeenCalled();
    expect(window.location.reload).toHaveBeenCalled();
    window.location.reload.mockRestore();

    done();
  });
});