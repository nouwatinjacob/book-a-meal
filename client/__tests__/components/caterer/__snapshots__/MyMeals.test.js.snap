// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyMeals component should render MyMeals component correctly 1`] = `
<div>
  <div>
    <CatererHeader />
    <h3
      className="text-center"
    >
      No Meals to display 
    </h3>
  </div>
</div>
`;
